File: Solutions/PS_0_solution/Cpp/Level_1.cpp
#include <iostream>
using namespace std;

int main(){
    int C;
    cin >> C;
    int n;
    cin >> n;

    // Make an array of weights, with size n. Making array of variable size is only allowed for gcc compiler (won't work in Clang or MSVC).
    int weight[n];
    for(int i=0; i<n; i++){
        cin >> weight[i];
    }

    // First one , total sum
    int total = 0;
    for(int i=0; i<n; i++){
        total += weight[i];
    }
    cout << "Total Shipment Weight:" << " " << total << endl;

    // Second one , avg. , we take float for decimals, and divide by 1.0 for float division.
    float avg = total/(1.0*n);
    cout << "Average Container Weight:" << " " << avg << endl;

    // Third and Fourth, we will do together. We pass one single loop, checking min and max;
    int minn = weight[0];
    int maxx = weight[0];
    for(int i=1; i<n; i++){
        if(weight[i] > maxx){
            maxx = weight[i];
        }
        if(weight[i] < minn){
            minn = weight[i];
        }
    }

    cout << "Heaviest Container:" << " " << maxx << endl;
    cout << "Lightest Container:" << " " << minn << endl;

    // Fifth one is just a single if statement
    if(total >=200){
        cout << "Classification:" << " " << "Heavy" << endl;
    }
    else{
        cout << "Classification:" << " " << "Light" << endl;
    }

    // Sixth one is direct output
    cout << "Port Capacity:" << " " << C << endl;

    //Last one is a if statement too
    if(total > C){
        cout << "Status:" << " " << "Shipment exceeds port capacity" << endl;
    }
    else{
        cout << "Status:" << " " << "Shipment can be unloaded" << endl;
    }
}

File: Solutions/PS_0_solution/Py/Level_1.py
#Read maximum storage capacity of the port
C = float(input())

#Read number of containers
N = int(input())

total = 0
heaviest = 0
lightest = 0

#Read weight of each container
for i in range(N):
    weight = float(input())

    #Calculate total shipment weight
    total += weight

    #Initialize heaviest and lightest using first container
    if i == 0:
        heaviest = weight
        lightest = weight

    #Find heaviest container
    if weight > heaviest:
        heaviest = weight

    #Find lightest container
    if weight < lightest:
        lightest = weight

#Calculate average container weight
average = total / N

print("Total Shipment Weight:", total)
print("Average Container Weight:", average)
print("Heaviest Container:", heaviest)
print("Lightest Container:", lightest)

#Classification is Heavy if total >= 200, otherwise Light
if total >= 200:
    print("Classification: Heavy")
else:
    print("Classification: Light")

#Print port capacity
print("Port Capacity:", C)

#Check if shipment can be unloaded
if total <= C:
    print("Status: Shipment can be unloaded")
else:
    print("Status: Shipment exceeds port capacity")
